docs/DeveloperGuide.md

```diff
@@ -4,35 +4,186 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+App initialisation: Initialises the various components, then begins reading inputs from the user
 
+![Architecture](UML-diagrams/Samuel/Architecture.png)
 
-## Product scope
-### Target user profile
+`Ui`: takes in user inputs from the CLI and sends it to the Logic component. Also handles all writing of messages to the user
 
-{Describe the target user profile}
+`Logic`: takes in string input from Ui, and parses it into commands to be executed on variables stored in the model. Calls Ui to provide the user with feedback that the action has or has not been completed.
 
-### Value proposition
+`Model`: Handles all the cached variables, i.e. the list of Bouquets created by the user, the list of available Flowers that the user can add to their Bouquets
 
-{Describe the value proposition: what problem does it solve?}
+`Storage`: Handles storing and reading from hardware, such as saving the list of bouquets created by the user so they can access it later.
 
-## User Stories
+## Implementation
+
+### Adding/Removing Bouquets
+
+![AddBouquets](UML-diagrams/Samuel/AddBouquets.png)
+To add a new bouquet to the list of Bouquets saved in the BouquetList variable, the user would have to type `add <bouquet name>` into the CLI, which is then handled by `Ui.getInput` and passed to `Parser`, which 
+creates a Command `Addcommand` with a new Bouquet of that name as an argument. This command is then sent back to `Florizz`, the main class, where it is then executed. 
+
+If a bouquet of the same name already exists in BouquetList, an exception will be thrown and the user will be prompted to choose another name. This is facilitated by overriding the .equals() method in the `Bouquet` class such that
+when two Bouquets are compared, .equals() returns true when they have the same name, regardless of which flowers are contained within each `Bouquet`. This decision was made mainly to facilitate `DeleteBouquetCommand`, 
+so we can simply call BouquetList.remove(BouquetToRemove) and the correct bouquet will be removed even if the BouquetToRemove is simply an empty `Bouquet` with the same name as the one in the list.
+
+After adding the `Bouquet` to `BouquetList`, the new `Bouquet` is then sent to `ui.printBouquetAdded`, where a confirmation message will be printed to the user. 
+
+Removing Bouquets follows a similar logic, just that instead of calling `BouquetList.add(Bouquet)`, executing `DeleteBouquetCommand` calls `BouquetList.remove(Bouquet)`, removing the `Bouquet` as mentioned above.
+
+### Flower Information Command
+
+`info <flowerName>` command prints information about the specified flower
+
+![Info Command Sequence Diagram](/docs/UML-diagrams/Ian/InfoCommandUML.png)
+
+Step 1: Flower information mechanism utilize the `parser` class to parse user command for a specific flower name inputted.
```

Comment:
The step numbers in the section below (Flower Remove Command) are in bold, should standardise the formatting?